Sources/PrimerSDK/Classes/Core/PCI/TokenizationService.swift

```diff
@@ -42,18 +45,76 @@ internal class TokenizationService: TokenizationServiceProtocol {
         let api: PrimerAPIClientProtocol = DependencyContainer.resolve()
         
         api.tokenizePaymentMethod(clientToken: clientToken, paymentMethodTokenizationRequest: request) { (result) in
-            DispatchQueue.main.async {
-                switch result {
-                case .failure:
+            switch result {
+            case .failure:
+                DispatchQueue.main.async {
                     onTokenizeSuccess(.failure( PrimerError.tokenizationRequestFailed ))
+                }
+                
+            case .success(let paymentMethodToken):
+                let settings: PrimerSettingsProtocol = DependencyContainer.resolve()
+                let state: AppStateProtocol = DependencyContainer.resolve()
+                
+                var isThreeDSEnabled: Bool = false
+                
+                if state.paymentMethodConfig?.paymentMethods?.filter({ ($0.options as? CardOptions)?.threeDSecureEnabled == true }).count ?? 0 > 0 {
+                    isThreeDSEnabled = true
+                }
+                
+                self.tokenizedPaymentMethodToken = paymentMethodToken
+                               
+//                #if canImport(Primer3DS)
```

Comment:
do we want to keep these commented out fields?

Reply:
Fixed